### PR TITLE
Relative L2 loss

### PR DESCRIPTION
**Pull request by mnabian**
# PhysicsNeMo Pull Request

## Description

Adds relative MES and relative L2 to `physicsnemo.metrics.general`, and gives mse/rmse an optional weights argument for a masked/weighted mean. Both relative metrics accept element weights (subsuming a validity mask and per-point weights) and a dim reduction, matching general.mse.

## Checklist

- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: physicsnemo/metrics/general/relative_error.py [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    Returns
+    -------
+    Tensor
+        ``sum(weights * err_sq) / max(sum(weights * tgt_sq), eps)`` reduced over
+        ``dim``.
+    """
+    if weights is not None:
+        w = weights.to(device=err_sq.device, dtype=err_sq.dtype)
+        err_sq = err_sq * w
+        tgt_sq = tgt_sq * w
```
[the remark this answers is not in the document]
Comment (mnabian) [reply physicsnemo/metrics/general/relative_error.py:77]: Fixed.

File: physicsnemo/metrics/general/relative_error.py [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+    -------
+    Tensor
+        Relative L2 error reduced over ``dim``.
+
+    Raises
+    ------
+    ValueError
+        If ``pred`` and ``target`` do not have the same shape.
+    """
+    return relative_mse(pred, target, dim=dim, eps=eps, weights=weights).sqrt()
```
[the remark this answers is not in the document]
Comment (mnabian) [reply physicsnemo/metrics/general/relative_error.py]: Fixed.

File: physicsnemo/metrics/general/mse.py [hunk 3]
```diff
@@ -23,7 +23,13 @@
 Tensor = torch.Tensor
 
 
-def mse(pred: Tensor, target: Tensor, dim: int = None) -> Union[Tensor, float]:
+def mse(
```
Comment (coreyjadams) [inline physicsnemo/metrics/general/mse.py:26]: PyTorch has https://docs.pytorch.org/docs/2.13/generated/torch.nn.functional.mse_loss.html, which is quite similar.  Can we document here in the docstring any relevant differences users should be aware of?

File: physicsnemo/metrics/general/relative_error.py [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+        )
+
+
+def relative_mse(
+    pred: Tensor,
+    target: Tensor,
+    *,
+    dim: int | tuple[int, ...] | None = None,
+    eps: float = 1e-8,
+    weights: Tensor | None = None,
```
Comment (coreyjadams) [inline physicsnemo/metrics/general/relative_error.py]: It's a minor nit, but can we put these in the same order in `relative_error` and `mse`?  [pred, target, dim, weights, eps]` makes sense but I care more about consistency than detailed order, honestly.

File: physicsnemo/metrics/general/relative_error.py [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+        raise ValueError(
+            "pred and target must have the same shape, got "
+            f"{tuple(pred.shape)} and {tuple(target.shape)}"
+        )
+
+
+def relative_mse(
+    pred: Tensor,
+    target: Tensor,
+    *,
```
Comment (coreyjadams) [inline physicsnemo/metrics/general/relative_error.py]: Why * here and not in mse?

**Review by coreyjadams — APPROVED** [review]
Approving to unblock; I left a few small comments.

File: physicsnemo/metrics/general/mse.py (hunk 3 above)
Comment (mnabian) [reply physicsnemo/metrics/general/mse.py:26]: Added some explanation in the docstrings.

File: physicsnemo/metrics/general/relative_error.py (hunk 4 above)
Comment (mnabian) [reply physicsnemo/metrics/general/relative_error.py]: Sure, fixed.

File: physicsnemo/metrics/general/relative_error.py (hunk 5 above)
Comment (mnabian) [reply physicsnemo/metrics/general/relative_error.py]: Removed it from here.
